src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -669,7 +702,7 @@ private static BinderCheckTypeResult CheckEqualArgTypesCore(IErrorContainer erro
 
             // EqualOp is only allowed on primitive types, polymorphic lookups, and control types.
             if (!(typeLeft.IsPrimitive && typeRight.IsPrimitive) && !(typeLeft.IsPolymorphic && typeRight.IsPolymorphic) && !(typeLeft.IsControl && typeRight.IsControl)
-                && !(typeLeft.IsPolymorphic && typeRight.IsRecord) && !(typeLeft.IsRecord && typeRight.IsPolymorphic))
+                && !(typeLeft.IsPolymorphic && typeRight.IsRecord) && !(typeLeft.IsRecord && typeRight.IsPolymorphic) && !(typeLeft.IsUnknown || typeRight.IsUnknown))
```

Comment:
>&& !(typeLeft.IsUnknown || typeRight.IsUnknown) [](http://example.com/codeflow?start=118&length=47)

how do we hit this?

Reply:
This is hit on equality operators, here, if one of the operands is unknown, we still don't want to give an error.
e.g. unknown = number => bool